MobilePayKit.podspec

```diff
@@ -1,21 +1,23 @@
+# frozen_string_literal: true
+
 Pod::Spec.new do |s|
   s.name          = 'MobilePayKit'
-  s.version       = '1.0.0'
+  s.version       = '0.0.1'
```

Comment:
Well this seems like a regression 😜

Reply:
It's not.

The `1.0.0` tag never existed so far, and the `podspec` was created by the team with this value in there… even before the pod was pushed anywhere (and in fact it's still not pushed to trunk nor our private spec repo, since this PR was needed for this to happen), and while the [only tag that has been created (manually) by the team was `0.0.1`](https://github.com/Automattic/MobilePay-Apple/tags). So this is just fixing the initial `podspec` they wrote to match the tag they created.

Reply:
Makes sense – but we're still not at `1.0.0` yet? This'll be an in-development dependency for a while yet?

Reply:
Yep I think it's still a `0.0.1` version as it's being worked on by the team. @momo-ozawa will probably confirm.

According to SemVer, a `0.0.1` is totally valid to be pushed to CocoaPods anyway, and just signals that "the API of this library can change at any point because it's in its early stage of development so we're not commited to have the API being stable for a major version just yet"

---

The next step once this PR is merged, is for the team to continue working on the codebase until they consider that  the API of the library is stable enough to make a first major `1.0.0` version, or even an intermediate `0.y.z` version beforehand, then they will:
 - Cut a `release/x.y.z` branch, update the `s.version` line in the `.podspec` from there, get it merged
 - Create a `x.y.z` GitHub release and git tag – which will trigger the publication to CocoaPods trunk

Which is why I used `0.0.1` here to match the current tag and let the team update that value only when they're ready to make a new version and decide which version number that next one should be.

---

PS: Note that merging this PR will _not_ publish the already-existing `0.0.1` tag – which as I pointed out in [this previous comment](https://github.com/Automattic/MobilePay-Apple/pull/19#issuecomment-942750596), won't be able to be pushed to trunk anyway, due to compilation warnings.